File: E-bank/src/main/java/com/ebank/model/Transaction.java
package com.ebank.model;


import com.ebank.enums.TransactionType;
import com.ebank.enums.CardType;
import com.ebank.enums.TransferType;
import lombok.*;
import jakarta.persistence.*;

import java.sql.Date;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Transaction {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Date transactionDate;
    private Double amount;
    @Enumerated(EnumType.STRING)
    private TransactionType transactionType;
    private String description;
    @Enumerated(EnumType.STRING)
    private TransferType transferType;
    private String targetAccountNumber;

    @ManyToOne
    private Account account;

    @ManyToOne
    private Beneficiary beneficiary;
}

File: Ebank-Frontend/src/app/model/transaction/transaction.ts
import { Account } from '../account/account';
import { Beneficiary } from '../beneficiary/beneficiary';
import {TransferType} from "../../enum/transfer-type";
import {TransactionType} from "../../enum/transaction-type";

export class Transaction {
  id: number;
  transactionDate: Date;
  amount: number;
  transactionType: TransactionType;
  description: string;
  transferType: TransferType;
  targetAccountNumber: string;
  account: Account;
  beneficiary: Beneficiary;

  constructor(
    id: number,
    transactionDate: Date,
    amount: number,
    transactionType: TransactionType,
    description: string,
    transferType: TransferType,
    targetAccountNumber: string,
    account: Account,
    beneficiary: Beneficiary
  ) {
    this.id = id;
    this.transactionDate = transactionDate;
    this.amount = amount;
    this.transactionType = transactionType;
    this.description = description;
    this.transferType = transferType;
    this.targetAccountNumber = targetAccountNumber;
    this.account = account;
    this.beneficiary = beneficiary;
  }
}
